org.hl7.fhir.r5/src/main/java/org/hl7/fhir/r5/context/BaseWorkerContext.java

```diff
@@ -440,7 +441,29 @@ protected void copy(BaseWorkerContext other) {
     }
   }
   
-  
+  @Override
+  public boolean isKnownIdentifierSystem(String system) {
+    if (identifierSystems == null) {
+      identifierSystems = new HashMap<>();
+      List<NamingSystem> nsl = systems.getList();
+      for (NamingSystem ns : nsl) {
+        if (ns.getKind().equals(NamingSystem.NamingSystemType.IDENTIFIER)) {
+          for (NamingSystemUniqueIdComponent nsc : ns.getUniqueId()) {
+            if (nsc.getPreferred() && nsc.getType().equals(NamingSystemIdentifierType.URI)) {
```

Comment:
Preferred are still valid, so this logic should be enhanced to allow for preferred vs non-proffered (with no preferred)